### ISSUE TITLE
Array.prototype.find is always shimmed

### ISSUE THREAD

**Issue by zerovox** [paulmillr/es6-shim#433]
```
if (![, 1].find(function (item, idx) { return idx === 0; })) {
  overrideNative(Array.prototype, 'find', ArrayPrototypeShims.find);
}
```

I believe this will shim find if the behaviour of find is correct. `[, 1].find(function (item, idx) { return idx === 0; })` should return `undefined` (assuming my understanding that `[, 1]` should act the same as `[undefined, 1]`, so we want to shim if it returns `1`. This means the negation is placed incorrectly.

**Comment by ljharb** [comment]
That's not entirely accurate - we want to shim if it reaches the iteration of index 1 - but you're correct that this seems wrong, because of the `undefined`.

I'll get a fix out for this (and `findIndex`) shortly.

**Comment by ljharb** [comment]
(note that this isn't breaking anything, it's just needlessly shimming)

**Comment by ljharb** [comment]
Published as v0.35.3.
